packages/xod-project/src/flatten.js

```diff
@@ -746,29 +752,31 @@ export const extractPatches = R.curry((project, leafPaths, prefix, boundValues,
  * @private
  * @function convertPatch
  * @param {Project} project
- * @param {Array<String>} impls
  * @param {Array<Array<Path, Patch>>} leafPatches
  * @param {Patch} patch
  * @returns {Patch}
  */
- // :: Project  -> String[] -> [[Path, Patch]] -> Patch -> Patch
-const convertPatch = R.curry((project, impls, leafPatches, patch) => R.ifElse(
-  Patch.hasImpls(impls),
-  R.identity,
-  (originalPatch) => {
-    const leafPatchPaths = R.pluck(0, leafPatches);
-    const flattenEntities = extractPatches(project, leafPatchPaths, null, {}, originalPatch);
-    const nodes = R.map(R.unnest, flattenEntities[0]);
-    const links = R.map(R.unnest, flattenEntities[1]);
-    const [newNodes, newLinks] = createCastNodes(leafPatches, nodes, links);
-
-    return R.compose(
-      explodeEither,
-      Patch.upsertLinks(newLinks),
-      Patch.upsertNodes(newNodes)
-    )(Patch.createPatch());
-  }
-)(patch));
+const convertPatch = def(
+  'convertPatch :: Project -> [Pair PatchPath Patch] -> Patch -> Patch',
+  (project, leafPatches, patch) =>
+    R.ifElse(
+      Patch.hasImpl,
+      R.identity,
```

Comment:
`R.ifElse` could be replaced with `R.unless` in this case.

Reply:
Done